File: app/models/canned_message.rb
# == Schema Information
#
# Table name: canned_messages
#
#  id         :integer          not null, primary key
#  name       :string(255)
#  subject    :string(255)
#  text       :string(255)
#  note_text  :string(255)
#  default_to :string(255)
#  default_cc :string(255)
#  created_at :datetime
#  updated_at :datetime
#

class CannedMessage < ActiveRecord::Base
  using_access_control

  validates :name, :presence => { message: "A message's name cannot be blank" }
  validates :name, uniqueness: true
  validates :subject, :presence => { message: "A message's subject cannot be blank" }
  validates :text, :presence => { message: "A message's text cannot be blank" }

end

File: app/assets/javascripts/models/canned_message.js

var attr = DS.attr;

App.CannedMessage = DS.Model.extend({

  name: attr(),
  subject: attr(),
  text: attr(),
  note_text: attr(),
  default_to: attr(),
  default_cc: attr(),
  include_summary: attr('string', { defaultValue: 'true' }),

  isEditing: false

});
